src/libraries/System.Drawing.Common/src/System/Drawing/DrawingComWrappers.cs

```diff
@@ -0,0 +1,283 @@
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Drawing
+{
+    /// <summary>
+    /// The ComWrappers implementation for System.Drawing.Common's COM interop usages.
+    ///
+    /// Supports IStream and IPicture COM interfaces.
+    /// </summary>
+    internal unsafe class DrawingComWrappers : ComWrappers
+    {
+        private static readonly ComInterfaceEntry* s_wrapperEntry = InitializeComInterfaceEntry();
+        internal static DrawingComWrappers Instance { get; } = new DrawingComWrappers();
+
+        private DrawingComWrappers() { }
+
+        internal static void CheckStatus(int result)
+        {
+            if (result != 0)
+            {
+                throw new ExternalException() { HResult = result };
+            }
+        }
+
+        private static ComInterfaceEntry* InitializeComInterfaceEntry()
+        {
+            GetIUnknownImpl(out IntPtr fpQueryInteface, out IntPtr fpAddRef, out IntPtr fpRelease);
+
+            IntPtr iStreamVtblRaw = RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IStreamVtbl), sizeof(IStreamVtbl));
+            IStreamVtbl.Fill((IStreamVtbl*)iStreamVtblRaw, fpQueryInteface, fpAddRef, fpRelease);
+
+            ComInterfaceEntry* wrapperEntry = (ComInterfaceEntry*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IStreamVtbl), sizeof(ComInterfaceEntry));
+            wrapperEntry->IID = Interop.Ole32.IStreamComWrapper.IID;
+            wrapperEntry->Vtable = iStreamVtblRaw;
+            return wrapperEntry;
+        }
+
+        protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count)
+        {
+            Debug.Assert(obj is Interop.Ole32.IStreamComWrapper);
+            Debug.Assert(s_wrapperEntry != null);
+
+            // Always return the same table mappings.
+            count = 1;
+            return s_wrapperEntry;
+        }
+
+        protected override object CreateObject(IntPtr externalComObject, CreateObjectFlags flags)
+        {
+            Debug.Assert(flags == CreateObjectFlags.None);
+
+            Guid pictureIID = IPicture.IID;
+            int hr = Marshal.QueryInterface(externalComObject, ref pictureIID, out IntPtr comObject);
+            if (hr == 0)
+            {
+                return new PictureWrapper(comObject);
+            }
+
+            throw new NotImplementedException();
+        }
+
+        protected override void ReleaseObjects(IEnumerable objects)
+        {
+            throw new NotImplementedException();
+        }
+
+        internal struct IUnknownVtbl
+        {
+            public IntPtr QueryInterface;
+            public IntPtr AddRef;
+            public IntPtr Release;
+        }
+
+        internal unsafe struct IStreamVtbl
+        {
+            public IUnknownVtbl IUnknownImpl;
+            public delegate* unmanaged<IntPtr, byte*, uint, uint*, Interop.HRESULT> Read;
```

Comment:
Nit: Having this strong typed struct is unnecessary overhead. It is cheaper to just assign the pointers via indices like how it is done in the updated sample: https://github.com/dotnet/samples/pull/4673/files#diff-185fa28a60762c0a2cb21823eb106462ea7fbb7044c2708e56c83dbe61383c37R23

Reply:
Thanks for the suggestion. I've removed the struct, changed this to a `static class` and use the pattern in the updated sample.